sebs/azure/azure.py

```diff
@@ -287,13 +247,109 @@ def publish_function(
 
     def update_function(self, function: Function, code_package: Benchmark):
 
+        assert code_package.has_input_processed
+
         # Mount code package in Docker instance
         container_dest = self._mount_function_code(code_package)
-        url = self.publish_function(function, code_package, container_dest, True)
+        function_url = self.publish_function(function, code_package, container_dest, True)
+
+        envs = {}
+        if code_package.uses_nosql:
+
+            nosql_storage = cast(CosmosDB, self._system_resources.get_nosql_storage())
 
-        trigger = HTTPTrigger(url, self.config.resources.data_storage_account(self.cli_instance))
-        trigger.logging_handlers = self.logging_handlers
-        function.add_trigger(trigger)
+            # If we use NoSQL, then the handle must be allocated
+            _, url, creds = nosql_storage.credentials()
+            db = nosql_storage.benchmark_database(code_package.benchmark)
+            envs["NOSQL_STORAGE_DATABASE"] = db
+            envs["NOSQL_STORAGE_URL"] = url
+            envs["NOSQL_STORAGE_CREDS"] = creds
+
+            for original_name, actual_name in nosql_storage.get_tables(
+                code_package.benchmark
+            ).items():
+                envs[f"NOSQL_STORAGE_TABLE_{original_name}"] = actual_name
```

Comment:
Why are env vars the best approach in this situation?

Reply:
@oanarosca For object storage buckets, we had to pass their names as part of the input. I think that using envs is a better choice, as we don't have to adjust the input of each invocation.

Do you see a better way of doing that?